src/app/crud/base.py

```diff
@@ -58,11 +58,23 @@ async def get_by(self, field_name: str, val: Union[str, int], strict: bool = Fal
             )
         return entry
 
+    async def get_all_by(self, field_name: str, val: Union[str, int], strict: bool = False) -> List[ModelType]:
+        statement = select(self.model).where(getattr(self.model, field_name) == val)  # type: ignore[var-annotated]
+        results = await self.session.exec(statement=statement)
+        entries = results.all()
+
+        if strict and not entries:
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND,
+                detail=f"Table {self.model.__name__} has no corresponding entry.",
+            )
+        return entries
+
```

Comment:
cf previous comment. Apart from the strict (which we can check with `len` here), you have everything in `fetch_all`